### PR TITLE
Bug fixes

### LINKED ISSUE #102: Unable to relaunch a workflow run
## Version
Tower CLI Version 0.3 (build 1feb5a7)
Tower Cloud Version 21.10.14_a26ab055

## Steps to reproduce
1. Launch a workflow run
2. Wait for the run to complete successfully
3. Obtain the workflow run id
4. Attempt to relaunch specifying the workflow run id `-i`

```
$ tw runs list                       

  Pipeline's runs at [community / showcase] workspace:

     ID             | Status    | Project Name                          | Run Name                 | Username          | Submit Date                   
    ----------------+-----------+---------------------------------------+--------------------------+-------------------+-------------------------------
     2WmmVWxmnKuvq7 | RUNNING   | nf-core/rnaseq                        | compassionate_bose       | evanfloden        | Tue, 9 Nov 2021 19:05:36 GMT  
     2LKVd4dMLH8igz | SUCCEEDED | nf-core/fetchngs                      | fabulous_brattain        | drpatelhh         | Tue, 9 Nov 2021 15:29:31 GMT
     
$ tw runs relaunch -i 2LKVd4dMLH8igz

 ERROR: Field 'sessionId' with value 'null' must not be null

```

### LINKED ISSUE #119: NullPointerException when running tw workspaces leave
```
$ env | grep "TOWER"
TOWER_ACCESS_TOKEN=XXXXXXXXXXXXXXXXXXXXXXX

$ tw --version
Tower CLI version 0.3 (build fbb1932)

$ tw workspaces leave
java.lang.NullPointerException
        at io.seqera.tower.cli.commands.workspaces.LeaveCmd.exec(LeaveCmd.java:42)
        at io.seqera.tower.cli.commands.AbstractApiCmd.call(AbstractApiCmd.java:284)
        at io.seqera.tower.cli.commands.AbstractApiCmd.call(AbstractApiCmd.java:45)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at io.seqera.tower.cli.Tower.main(Tower.java:87)
```

Let me know if you need any furter info.

### LINKED ISSUE #122: No default options specified when running "tw participants leave"
Running the command without any options tells you that you are missing an option but no description as to what format it should be:

```
$ tw participants leave

 ERROR: Missing workspace option.
```

Running a similar command I get this which may be the expected behaviour?

```
$ tw members leave

Missing required option: '--organization=<organizationName>'

Usage: tw members leave [OPTIONS]

Leave an organization

Options:
* -o, --organization=<organizationName>   Organization's name identifier
  -h, --help                              Show this help message and exit.
  -V, --version                           Print version information and exit.
```

### PR DESCRIPTION
**Pull request by jordeu**
- Make workspace required at all participants subcommands. Fixes #122 #118
- Use `WorkspaceRequiredOptions` at all workspaces commands. Fixes #119 
- Fix workspace update command
- Fix relaunch with resume. Fixes #102

### CODE REVIEW

**Review by swampie — APPROVED** [review]

File: src/main/java/io/seqera/tower/cli/commands/global/WorkspaceRequiredOptions.java [hunk 1]
```diff
@@ -13,8 +13,8 @@
 
 import picocli.CommandLine;
 
-public class WorkspaceOptions {
+public class WorkspaceRequiredOptions {
 
-    @CommandLine.Option(names = {"-w", "--workspace"}, description = "Workspace numeric identifier (TOWER_WORKSPACE_ID)", defaultValue = "${TOWER_WORKSPACE_ID}")
+    @CommandLine.Option(names = {"-w", "--workspace"}, description = WorkspaceOptionalOptions.DESCRIPTION, defaultValue = WorkspaceOptionalOptions.DEFAULT_VALUE, required = true)
```
Comment (grananda) [inline src/main/java/io/seqera/tower/cli/commands/global/WorkspaceRequiredOptions.java:18]: Could we try using the same convention as for `WorkspaceOptionalOptions` when it comes to `description` and `defaultValue`? (I meant, using constants)

**Review by grananda — APPROVED** [review]

**Comment by grananda** [comment]
I see you have come ahead and started replacing the resources definition by Id only, at least when it comes to WSP. Good.

File: src/main/java/io/seqera/tower/cli/commands/global/WorkspaceRequiredOptions.java (hunk 1 above)
Comment (jordeu) [reply src/main/java/io/seqera/tower/cli/commands/global/WorkspaceRequiredOptions.java:18]: I don't get what do you mean. Here I'm using the `WorkspaceOptionalOptions` constants to keep both in sync.

Comment (grananda) [reply src/main/java/io/seqera/tower/cli/commands/global/WorkspaceRequiredOptions.java:18]: Sorry, you`re right. I focused on the missing constant and missed the static reference to them.

**Pull request merged**